home_assistant_streamdeck_yaml.py

```diff
@@ -62,9 +64,281 @@
+        default=None,
+        allow_template=True,
+    )
+    linked_entity: str | None = Field(
+        default=None,
+        allow_template=True,
+        description="A secondary entity_id that is used for updating images and states"
+    )
+    service: str | None = Field(
+        default=None,
+        allow_template=True,
+    )  
+    service_data: dict[str,Any] | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    target: dict[str,Any] | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    dial_event_type: str | None = Field(
+        default=None,
+        allow_template=True,
+        description="The event type of the dial that will trigger the service."
+        "Either DialEventType.TURN or DialEventType.PUSH"
+    )
+    text: str = Field(
+        default="",
+        allow_template=True
+    )
+    text_color: str | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    text_size: int = Field(
+        default=16,
+        allow_template=False,
+    )
+    text_offset: int = Field(
+        default=0,
+        allow_template=False,
+    )
+    icon: str | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    icon_mdi: str | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    icon_background_color: str = Field(
+        default="#000000",
+        allow_template=True,
+    )
+    icon_mdi_color: str | None = Field(
+        default=None,
+        allow_template=True,
+    )
+    icon_gray_when_off: bool = Field(
+        default=False,
+        allow_template=False,
+        description="When specifying `icon` and `entity_id`, if the state is `off`, the icon will be converted to grayscale.",
+    )
+    delay: float | str = Field(
+        default=0.0,
+        allow_template=True,
+        description="The delay inbetween events for the service to be called"
+        " Dial changes are added to decrease traffic ",
+    )
+    state_attribute: str | None = Field(
+        default=None,
+        allow_template=True,
+        description="The attribute of the entity which gets used for the dial state"
+    )
+    attributes: dict[str, Any] | None = Field(
+        default=None,
+        allow_template=True,
+        description="Sets the attributes of the dial"
+        "min: The minimal value of the dial"
+        "max: The maximal value of the dial"
+        "step: the step size by which the value of the dial is increased by on an event",
+    )
+    allow_touchscreen_events: bool = Field(
+        default=False,
+        allow_template=True,
+        description="Whether events from the touchscreen such as setting minimal value on short and setting maximal value on LONG"
+    )
+
+    _min : float = PrivateAttr(0)
+    _max : float = PrivateAttr(100)
+    _step: float = PrivateAttr(1)
+    _state: float = PrivateAttr(0)
+    
+    def update_attributes(self, data: dict[str, Any]) -> None:
```

Comment:
your try/catch seems like a bad idea, f.e. if min is not set, max and step will not be evaluated anymore. Is this intended behavior? Why do we need the `if self.attributes is None:` block at all?  
If you really need to get from `data["attributes"]` or `self.attributes` better use something like this:

```python
attributes = self.attributes or data["attributes"]
```